pkg/utils/uuid_test.go

```diff
@@ -31,6 +31,54 @@ func TestFakeUUID(t *testing.T) {
 	tests.Assert(t, uuid == "68656b65746968656b65746968656b65")
 }
 
+func TestNonRandomUUID(t *testing.T) {
+	s := IdSource{&NonRandom{}}
+
+	uuid := s.ReadUUID()
+	tests.Assert(t, len(uuid) == 32, "expected len(uuid) == 32, got:", len(uuid))
+	tests.Assert(t, uuid == "00000000000000000000000000000000", "got:", uuid)
+	uuid = s.ReadUUID()
+	tests.Assert(t, len(uuid) == 32, "expected len(uuid) == 32, got:", len(uuid))
+	tests.Assert(t, uuid == "00000000000000000000000000000001", "got:", uuid)
+	uuid = s.ReadUUID()
+	tests.Assert(t, len(uuid) == 32, "expected len(uuid) == 32, got:", len(uuid))
+	tests.Assert(t, uuid == "00000000000000000000000000000002", "got:", uuid)
+
+	for i := 0; i < 106; i++ {
+		s.ReadUUID()
+	}
+	uuid = s.ReadUUID()
+	tests.Assert(t, len(uuid) == 32, "expected len(uuid) == 32, got:", len(uuid))
+	tests.Assert(t, uuid == "0000000000000000000000000000006d", "got:", uuid)
+}
+
+func TestReplaceRandomness(t *testing.T) {
+	before := Randomness
+
+	n := &NonRandom{}
+	Randomness = n
+	defer func() {
+		Randomness = before
+	}()
+
+	// now were using a non-random source. we should have predictible values
+	uuid := GenUUID()
+	tests.Assert(t, len(uuid) == 32, "expected len(uuid) == 32, got:", len(uuid))
+	tests.Assert(t, uuid == "00000000000000000000000000000000", "got:", uuid)
+
+	uuid = GenUUID()
+	tests.Assert(t, len(uuid) == 32, "expected len(uuid) == 32, got:", len(uuid))
+	tests.Assert(t, uuid == "00000000000000000000000000000001", "got:", uuid)
+
+	// restore the random source back to what it was before
+	Randomness = before
+
+	// uuid should NOT be the next non-random number
+	uuid = GenUUID()
+	tests.Assert(t, len(uuid) == 32, "expected len(uuid) == 32, got:", len(uuid))
+	tests.Assert(t, uuid != "00000000000000000000000000000001", "got:", uuid)
```

Comment:
I think the next would be `000....02` not `0000...01`

Reply:
Wow. Good catch. That was a copy and paste error. I've now proved that is fails the test when I comment out the line that restores the previous version of Randomness.